### PR TITLE
rtorrent-ps, libtorrent-ps: init at 1.1-37c9d4b

### PR DESCRIPTION
**Pull request by yorickvP**
###### Motivation for this change

[rtorrent-ps](https://github.com/pyroscope/rtorrent-ps/) is a popular patch set for rtorrent. These packages are inspired by the [aur PKGBUILD](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=rtorrent-ps)

(cc: @pyroscope, @ebzzry, @codyopel)

I also have a pyrocore (python) package that should be upstreamed at some point.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by yorickvP** [comment]
I sitll want this.

**Comment by yorickvP** [comment]
I still want this.

File: pkgs/tools/networking/p2p/libtorrent/default.nix [hunk 1]
```diff
@@ -15,10 +15,8 @@ stdenv.mkDerivation rec {
     sha256 = "027qanwcisxhx0bq8dn8cpg8563q0k2pm8ls278f04n7jqvvwkp0";
   };
 
-  nativeBuildInputs = [ pkgconfig ];
-  buildInputs = [ libtool autoconf automake cppunit openssl libsigcxx zlib ];
-
-  preConfigure = "./autogen.sh";
+  nativeBuildInputs = [ autoreconfHook pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/p2p/libtorrent/default.nix:18]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config ];
```

File: pkgs/tools/networking/p2p/libtorrent/ps.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      owner = "rakshasa";
+      repo = "libtorrent";
+      rev = "${version}";
+      sha256 = "1rvrxgb131snv9r6ksgzmd74rd9z7q46bhky0zazz7dwqqywffcp";
+    };
+  });
+in
+libtorrent_0_13_6.overrideAttrs (o: rec {
+  version = "1.1-${builtins.substring 0 7 ps.rev}";
+  name = "libtorrent-ps-${version}";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/p2p/libtorrent/ps.nix:22]:
```suggestion
  pname = "libtorrent-ps";
```

File: pkgs/tools/networking/p2p/libtorrent/ps.nix [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  buildInputs = (stdenv.lib.remove openssl o.buildInputs) ++ [ openssl_1_1 ];
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/p2p/libtorrent/ps.nix:23]:
```suggestion
  buildInputs = (lib.remove openssl o.buildInputs) ++ [ openssl_1_1 ];
```

File: pkgs/tools/networking/p2p/libtorrent/ps.nix [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    "backport*_${o.version}_*.patch"
+    "backport*_all_*.patch"
+    "lt-base-cppunit-pkgconfig.patch" # 0.13.6
+    "lt-open-ssl-1.1.patch"
+    "lt-ps-*_${o.version}.patch"
+    "lt-ps-*_all.patch"
+  ]);
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/p2p/libtorrent/ps.nix:34]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/networking/p2p/rtorrent/default.nix [hunk 5]
```diff
@@ -18,21 +15,11 @@ stdenv.mkDerivation rec {
     sha256 = "0a9dk3cz56f7gad8ghsma79iy900rwdvzngs6k6x08nlwaqid8ga";
   };
 
-  nativeBuildInputs = [ pkgconfig ];
+  nativeBuildInputs = [ autoreconfHook pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/p2p/rtorrent/default.nix:18]:
```suggestion
  nativeBuildInputs = [ autoreconfHook pkg-config ];
```

File: pkgs/tools/networking/p2p/rtorrent/ps.nix [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      rev = "${version}";
+      sha256 = "0iyxmjr1984vs7hrnxkfwgrgckacqml0kv4bhj185w9bhjqvgfnf";
+    };
+  });
+in
+
+rtorrent_0_9_6.overrideAttrs (o: rec {
+  version = "1.1-${builtins.substring 0 7 ps.rev}";
+  name = "rtorrent-ps-${version}";
+  buildInputs = (stdenv.lib.subtractLists [ libtorrent openssl ] o.buildInputs) ++ [ libtorrent-ps openssl_1_1 ];
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/p2p/rtorrent/ps.nix:24]:
```suggestion
  buildInputs = (lib.subtractLists [ libtorrent openssl ] o.buildInputs) ++ [ libtorrent-ps openssl_1_1 ];
```

File: pkgs/tools/networking/p2p/rtorrent/ps.nix [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      repo = "rtorrent";
+      rev = "${version}";
+      sha256 = "0iyxmjr1984vs7hrnxkfwgrgckacqml0kv4bhj185w9bhjqvgfnf";
+    };
+  });
+in
+
+rtorrent_0_9_6.overrideAttrs (o: rec {
+  version = "1.1-${builtins.substring 0 7 ps.rev}";
+  name = "rtorrent-ps-${version}";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/p2p/rtorrent/ps.nix:23]:
```suggestion
  pname = "rtorrent-ps";
```

File: pkgs/tools/networking/p2p/rtorrent/ps.nix [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    ln -s ${ps}/patches/*.{cc,h} src/
+    RT_HEX_VERSION=$(printf "0x%02X%02X%02X" ${stdenv.lib.replaceChars ["."] [" "] o.version})
+    sed -i -e "s:\\(AC_DEFINE(HAVE_CONFIG_H.*\\):\1  AC_DEFINE(RT_HEX_VERSION, "$RT_HEX_VERSION", for CPP if checks):" configure.ac
+    grep "AC_DEFINE.*API_VERSION" configure.ac >/dev/null || sed -i -e "s:\\(AC_DEFINE(HAVE_CONFIG_H.*\\):\1  AC_DEFINE(API_VERSION, 0, api version):" configure.ac
+    sed -i -e 's/rTorrent \" VERSION/rTorrent PS-'"${version}"' " VERSION/' src/ui/download_list.cc
+  '';
+  
+  enableParallelBuilding = true;
+  
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/p2p/rtorrent/ps.nix:45]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
> I still want this.

It would be great if you would make sure that the PR cleanly merged and evals before writing this.

**Comment by yorickvP** [comment]
Superseded by #223712

**Pull request closed without merging**